coalib/results/result_actions/ShowPatchAction.py

```diff
@@ -80,18 +80,22 @@ def apply(self,
         printer = ConsolePrinter(colored)
 
         for filename, this_diff in sorted(result.diffs.items()):
+            to_filename = this_diff.rename if this_diff.rename else filename
+            to_filename = "/dev/null" if this_diff.delete else to_filename
             original_file = original_file_dict[filename]
             try:
                 current_file = file_diff_dict[filename].modified
-                new_file = (file_diff_dict[filename] + this_diff).modified
+                new_file = ([] if this_diff.delete
+                            else
+                            (file_diff_dict[filename] + this_diff).modified)
```

Comment:
actually modified should return `[]` on itself if the file has the deletion flag set so the if else is not needed